docker/pandoc-lambda/function/app.py

```diff
@@ -123,6 +127,166 @@ def promote_case_footnotes(doc):
+                cols_element, 'w:pgNumType', 'w:formProt', 'w:vAlign', 'w:noEndnote', 'w:titlePg',
+                'w:textDirection', 'w:bidi', 'w:rtlGutter', 'w:docGrid',
+                'w:printerSettings', 'w:sectPrChange'
+            )
+
+    # the number of body styles we use consistently is much smaller than the number of section topper styles, but the
+    # values are more predictable. Body text can get unexpected style names handed down from pandoc
+    topper_styles = [ 'About Page Title', 'Acknowledgements Subtitle', 'Acknowledgements Title',
+                      'Node End', 'Node Start',
+                      'Head Separator', 'Head End', 'Head Field Separator',
+                      'Casebook Headnote', 'Casebook Headnote Title', 'Casebook Link', 'Casebook Number',
+                      'Casebook Subtitle', 'Casebook Title',
+                      'Chapter Headnote', 'Chapter Link', 'Chapter Number', 'Chapter Subtitle', 'Chapter Title', 'Chapter Spacer',
+                      'Section Headnote', 'Section Link', 'Section Number', 'Section Subtitle', 'Section Title',
+                      'Resource Headnote', 'Resource Link', 'Resource Number', 'Resource Subtitle', 'Resource Title',
+                      'Case Header']
+                      # 'Heading 1', 'Heading 2', 'Heading 3', 'Heading 4', 'Heading 5', 'Heading 6', 'Heading 7',
+                      # 'Heading 8', 'Heading 9',
+                      # 'Subheading 1', 'Subheading 2', 'Subheading 3', 'Subheading 4', 'Subheading 5', 'Subheading 6',
+                      # 'Subheading 7', 'Subheading 8', 'Subheading 9']
+
+    section_break("doc-wide") # set the values in the section-wide sectpr
+
+    # this is like twice as fast as navigating the xml or the docx object
+    grafs = [p.style.name for p in doc.paragraphs]
+    for i, p in enumerate(grafs):
+        if p == 'Front Matter End':
+            section_break(doc.paragraphs[i], frontmatter=True)
+        elif p == 'Node End' and (grafs[i - 1] not in topper_styles):
+            section_break(doc.paragraphs[i], paragraph=True)
+        elif p == 'Node End':
+            section_break(doc.paragraphs[i])
+        elif p == 'Head End':
+            if grafs[i - 1].startswith('Chapter'):
+                section_break(doc.paragraphs[i-1], chapter=True)
+            if grafs[i - 1].startswith('Section'):
+                section_break(doc.paragraphs[i-1], section=True)
+            if grafs[i + 1] == 'Case Header':
+                continue
+            else:
+                section_break(doc.paragraphs[i])
+        elif p == 'Case Header':
+            if grafs[i + 1] == 'Case Header':
+                continue
+            else:
+                section_break(doc.paragraphs[i])
+        elif p == 'Case Body':
+            if grafs[i + 1] not in topper_styles:
+                continue
+            else:
+                section_break(doc.paragraphs[i], paragraph=True)
+        #elif p.startswith()
```

Comment:
to be removed?

Reply:
Yeah— this is all different with cols removed.